src/uu/ls/src/ls.rs

```diff
@@ -1411,6 +1411,1328 @@ only ignore '.' and '..'.",
         )
 }
 
+/// The entry point for the dir coreutil
+pub fn dir_main(args: impl uucore::Args) -> UResult<()> {
+    let command = dir_uu_app();
+    let matches = command.get_matches_from(args);
+
+    let mut default_quoting_style = false;
+    let mut default_format_style = false;
+
+    // We check if any options on formatting or quoting style have been given.
+    // If not, we will use dir default formatting and quoting style options
+
+    if !matches.is_present(options::QUOTING_STYLE)
+        && !matches.is_present(options::quoting::C)
+        && !matches.is_present(options::quoting::ESCAPE)
+        && !matches.is_present(options::quoting::LITERAL)
+    {
+        default_quoting_style = true;
+    }
+    if !matches.is_present(options::FORMAT)
+        && !matches.is_present(options::format::ACROSS)
+        && !matches.is_present(options::format::COLUMNS)
+        && !matches.is_present(options::format::COMMAS)
+        && !matches.is_present(options::format::LONG)
+        && !matches.is_present(options::format::LONG_NO_GROUP)
+        && !matches.is_present(options::format::LONG_NO_OWNER)
+        && !matches.is_present(options::format::LONG_NUMERIC_UID_GID)
+        && !matches.is_present(options::format::ONE_LINE)
+    {
+        default_format_style = true;
+    }
+
+    let mut config = Config::from(&matches)?;
+
+    if default_quoting_style {
+        config.quoting_style = QuotingStyle::C {
+            quotes: Quotes::None,
+        };
+    }
+    if default_format_style {
+        config.format = Format::Columns;
+    }
+
+    let locs = matches
+        .values_of_os(options::PATHS)
+        .map(|v| v.map(Path::new).collect())
+        .unwrap_or_else(|| vec![Path::new(".")]);
+    list(locs, &config)
+}
+
+fn dir_uu_app<'a>() -> Command<'a> {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .override_usage(format_usage(USAGE))
+        .about(
+            "dir is a shortcut for ls -C -b. All valid ls options are also valid dir options",
+        )
+        .infer_long_args(true)
+        .arg(
+            Arg::new(options::HELP)
+                .long(options::HELP)
+                .help("Print help information.")
+        )
+        // Format arguments
+        .arg(
```

Comment:
could you please refactor this ? it is a lot of duplicate content :)

Reply:
No problem

Reply:
> could you please refactor this ? it is a lot of duplicate content :)

Done